pkg/cmd/operator/probeserver/scylladbapistatus.go

```diff
@@ -0,0 +1,179 @@
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/klog/v2"
+)
+
+type ScyllaDBAPIStatusOptions struct {
+	ServeProbesOptions
+
+	genericclioptions.ClientConfig
+	genericclioptions.InClusterReflection
+	ServiceName string
+
+	mux        *http.ServeMux
+	kubeClient kubernetes.Interface
+}
+
+func NewScyllaDBAPIStatusOptions(streams genericclioptions.IOStreams) *ScyllaDBAPIStatusOptions {
+	mux := http.NewServeMux()
+
+	return &ScyllaDBAPIStatusOptions{
+		ServeProbesOptions: *NewServeProbesOptions(streams, naming.ScyllaDBAPIStatusProbePort, mux),
+		ClientConfig:       genericclioptions.NewClientConfig("scylla-operator-api-status-probe"),
+		mux:                mux,
+	}
+}
+
+func (o *ScyllaDBAPIStatusOptions) AddFlags(cmd *cobra.Command) {
+	o.ServeProbesOptions.AddFlags(cmd)
+	o.ClientConfig.AddFlags(cmd)
+	o.InClusterReflection.AddFlags(cmd)
+
+	cmd.Flags().StringVarP(&o.ServiceName, "service-name", "", o.ServiceName, "Name of the service corresponding to the managed node.")
+}
+
+func NewScyllaDBAPIStatusCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewScyllaDBAPIStatusOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "scylladb-api-status",
+		Short: "Serves probes based on ScyllaDB API status.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(streams, cmd)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *ScyllaDBAPIStatusOptions) Validate(args []string) error {
+	var errs []error
+
+	errs = append(errs, o.ServeProbesOptions.Validate(args))
+	errs = append(errs, o.ClientConfig.Validate())
+	errs = append(errs, o.InClusterReflection.Validate())
+
+	if len(o.ServiceName) == 0 {
+		errs = append(errs, fmt.Errorf("service-name can't be empty"))
+	} else {
+		serviceNameValidationErrs := apimachineryvalidation.NameIsDNS1035Label(o.ServiceName, false)
+		if len(serviceNameValidationErrs) != 0 {
+			errs = append(errs, fmt.Errorf("invalid service name %q: %v", o.ServiceName, serviceNameValidationErrs))
+		}
+	}
+
+	return apierrors.NewAggregate(errs)
+}
+
+func (o *ScyllaDBAPIStatusOptions) Complete(args []string) error {
+	err := o.ServeProbesOptions.Complete(args)
+	if err != nil {
+		return err
+	}
+
+	err = o.ClientConfig.Complete()
+	if err != nil {
+		return err
+	}
+
+	err = o.InClusterReflection.Complete()
+	if err != nil {
+		return err
+	}
+
+	o.kubeClient, err = kubernetes.NewForConfig(o.ProtoConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes clientset: %w", err)
+	}
+
+	return nil
+}
+
+func (o *ScyllaDBAPIStatusOptions) Run(originalStreams genericclioptions.IOStreams, cmd *cobra.Command) (returnErr error) {
+	klog.Infof("%s version %s", cmd.Name(), version.Get())
+	cliflag.PrintFlags(cmd.Flags())
+
```

Comment:
Other exported Run method doesn't print it. Maybe the other one doesn't need to be exported at all?

Reply:
it's public so it can be used by different command that build on top of this, while it can print its own "header"

Reply:
Run can also be reused and it has a printing side effect while other doesn't. From function signature it's not clear which one is silent and which not. 
Maybe printing in RunE would be better a place.

Reply:
You get the choice of which one to call, inlining it into RunE (aside from making it bigger) would take it away.

> From function signature it's not clear which one is silent and which not.

The one with context assumes it's not the entrypoint (otherwise it would have to setup the cancel signal), the one with context assume it's handled by the caller.

Reply:
I'm with with at least a comment explaining which one prints and which not.

Reply:
I have renamed the methods. Hopefully, the distinction is clear now.